epaint/src/shape.rs

```diff
@@ -464,3 +479,902 @@ fn dashes_from_line(
+        });
+        
+        assert_eq!(result.len(), 100);
+
+        let curve = CubicBezierShape {
+            points: [
+                Pos2::new(90.0, 110.0),
+                Pos2::new(110.0, 170.0),
+                Pos2::new(210.0, 70.0),
+                Pos2::new(170.0, 110.0),
+            ],
+            closed: false,
+            fill: Default::default(),
+            stroke: Default::default(),
+        };
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(0.01, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 71);
+
+        let curve = CubicBezierShape {
+            points: [
+                Pos2::new(90.0, 110.0),
+                Pos2::new(110.0, 170.0),
+                Pos2::new(150.0, 50.0),
+                Pos2::new(170.0, 110.0),
+            ],
+            closed: false,
+            fill: Default::default(),
+            stroke: Default::default(),
+        };
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(0.01, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 88);
+    }
+
+    #[test]
+    fn test_quadrtic_flattening() {
+        let curve = QuadraticBezierShape {
+            points: [
+                Pos2::new(0.0, 0.0),
+                Pos2::new(80.0, 200.0),
+                Pos2::new(100.0, 30.0),
+            ],
+            closed: false,
+            fill: Default::default(),
+            stroke: Default::default(),
+        };
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(1.0, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 9);
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(0.5, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 11);
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(0.1, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 24);
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(0.01, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 72);
+
+        let mut result = Vec::new();
+        result.push(curve.points[0]); //add the start point
+        curve.for_each_flattened_with_t(0.001, &mut |pos,_t| {
+            result.push(pos);
+        });
+        
+        assert_eq!(result.len(), 223);
+    }
+}
```

Comment:
There is a lot of new code here - I think it all in its own file (`bezier.rs`)

Reply:
yes, moved all Cubic and Quadratic code to a new 'bezier.rs' file.